tools/clang/tools/libclang/dxcrewriteunused.cpp

```diff
@@ -54,19 +55,21 @@ using namespace llvm;
 using namespace clang;
 using namespace hlsl;
 
-class RewriteUnusedASTConsumer : public SemaConsumer {
-private:
-  Sema* m_sema = nullptr;
-public:
-  RewriteUnusedASTConsumer() {
-  }
-  void InitializeSema(Sema& S) override {
-    m_sema = &S;
-  }
-  void ForgetSema() override {
-    m_sema = nullptr;
-  }
+struct RewriteHelper {
+  SmallPtrSet<VarDecl *, 128> unusedGlobals;
+  SmallPtrSet<FunctionDecl *, 128> unusedFunctions;
+  SmallPtrSet<TypeDecl *, 32> unusedTypes;
+  DenseMap<RecordDecl *, unsigned> anonymousRecordRefCounts;
 };
+
+struct ASTHelper {
+  CompilerInstance compiler;
+  TranslationUnitDecl *tu;
+  ParsedSemanticDefineList macros;
```

Comment:
The name `macros` does not make it clear that these are the semantic defines, not regular macros.  Only when seen next to userMacros is it more clear.